backend/de.metas.marketing/base/src/main/java/de/metas/marketing/base/interceptor/MKTG_Channel.java

```diff
@@ -54,20 +58,30 @@ private MKTG_Channel()
 			timings = { ModelValidator.TYPE_BEFORE_DELETE })
 	public void checkIfCanBeDeleted(@NonNull final I_MKTG_Channel mktgChannel)
 	{
+		if (!isMarketingChannelsUseEnforced())
```

Comment:
But can we generally delete a marketingchannel just like that? wont there be a DB exception in case the channel is still referenced by `AD_User_MKTG_Channel` records?
So i think you probably don't need to change this method at all..

Reply:
@metas-ts clarified with @metas-mk It's ok to not be able to delete a marketing channel if a there is still a reference on the mktg_channel_id from the AD_User_MKTG_Channels table. This means we could completely delete this interceptor and the de.metas.marketing.base.api.IMKTGChannelDao#retrieveUsersHavingChannel method. However, AFAIU the possibility to only not be able to remove a channel if a user would remain with 0 will be implemented in another task, so I propose to comment out the line (+a comment with this task and explanation) that registers this interceptor and expand this in the future (A IMktgChannelBL class with a method that does necessary operation on the ad_user_mktg_channels table. WDYT?

Reply:
@dragospodariu96 

> It's ok to not be able to delete a marketing channel if a there is still a reference on the mktg_channel_id from the AD_User_MKTG_Channels table. 

OK

>  This means we could completely delete this interceptor 

no, because we still want a user-friendly error message when they try to delete a `MKTG_Channel` that is still linked to users.
that's why i proposed to "revert" it.

> will be implemented in another task[...](+a comment with this task and explanation)

if you think it will help in future, feel free to add a comment or something, so a future dev doesn't have to start at zero

Reply:
@metas-ts 

 Updated the interceptor's method. User friendly error message now shown when trying to delete a marketing channel that is still assigned to at least one user, regardless of the value from the 'enforce marketing channels use' sys config. 

OBS: Because regardless of that config, if we tried to delete a channel still assigned to a user, we would get an error due to the foreign key constraint on the ad_user_mktg_channels table